tracer/src/Datadog.Trace/Logging/Internal/DatadogLogging.cs

```diff
@@ -218,5 +228,39 @@ internal static string GetLogDirectory()
 
             return logDirectory;
         }
+
+        internal static void CleanLogFiles(string logsDirectory)
+        {
+            var logDaysLimit = EnvironmentHelpers.GetEnvironmentVariable(ConfigurationKeys.MaxLogFilesDays);
+            if (int.TryParse(logDaysLimit, out var days) && (days > 0 && days < int.MaxValue))
+            {
+                var date = DateTime.Now.AddDays(-1 * days);
+                var logFormats = new[]
+                  {
+                    "dotnet-tracer-managed-*-.log",
+                    "dotnet-tracer-native-*-.log",
+                    "dotnet-native-loader-*.log",
+                    "DD-DotNet-Profiler-Native-*.log"
+                  };
+
+                foreach (var logFormat in logFormats)
+                {
+                    foreach (var logFile in Directory.EnumerateFiles(logsDirectory, logFormat))
```

Comment:
This enumerates the directory 4 times. I wonder if it would be better to enumerate _once_ and check each file for the expected prefixes. Probably not (and more complex), just thinking out loud